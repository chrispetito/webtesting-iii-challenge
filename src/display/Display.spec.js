// Test away!
import React from "react";
import { render, toHaveClass } from "@testing-library/react";
import "@testing-library/react/cleanup-after-each";
import Display from "./Display";
import "jest-dom/extend-expect";

describe("<Display />", () => {
  it("display renders without crashing", () => {
    render(<Display />);
  });
  it("displays Closed and Locked when closed and locked are true", () => {
    const { getByTestId } = render(<Display closed={true} locked={true} />);
    const closed = getByTestId("closed");
    const locked = getByTestId("locked");
    expect(closed.textContent).toBe("Closed");
    expect(locked.textContent).toBe("Locked");
  });
  it("displays Open and Unlocked when closed and locked are false", () => {
    const { getByTestId } = render(<Display closed={!true} locked={!true} />);
    const closed = getByTestId("closed");
    const locked = getByTestId("locked");
    expect(closed.textContent).toBe("Open");
    expect(locked.textContent).toBe("Unlocked");
  });
  it("uses the red-led class when locked and closed are true", () => {
    const { getByTestId } = render(<Display closed={true} locked={true} />);
    const closed = getByTestId("closed");
    const locked = getByTestId("locked");
    expect(locked).toHaveClass("red-led");
    expect(closed).toHaveClass("red-led");
  });
  it("uses the green-led class when unlocked and closed are true", () => {
    const { getByTestId } = render(<Display closed={!true} locked={!true} />);
    const closed = getByTestId("closed");
    const locked = getByTestId("locked");
    expect(closed).toHaveClass("green-led");
    expect(locked).toHaveClass("green-led");
  });
});
