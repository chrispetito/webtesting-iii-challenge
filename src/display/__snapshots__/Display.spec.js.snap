// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> display matches snapshot 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="locked"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="closed"
    >
      Open
    </div>
  </div>
</div>
`;
